// material-ui-responsive-drawer/lib/store/actions.js
'use strict';

exports.__esModule = true;
exports.toggleDrawerOpen = toggleDrawerOpen;
exports.toggleDrawerDock = toggleDrawerDock;
exports.setDrawerOpen = setDrawerOpen;
exports.setResponsive = setResponsive;

var _types = require('./types');

var types = _interopRequireWildcard(_types);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function toggleDrawerOpen() {
  return {
    type: types.RESPONSIVE_DRAWER_TOGGLE_DRAWER_OPEN
  };
}

function toggleDrawerDock() {
  return {
    type: types.RESPONSIVE_DRAWER_TOGGLE_DRAWER_DOCK
  };
}

function setDrawerOpen(open) {
  return {
    type: types.RESPONSIVE_DRAWER_SET_DRAWER_OPEN,
    open: open
  };
}

function setResponsive(responsive) {
  return {
    type: types.RESPONSIVE_DRAWER_SET_RESPONSIVE,
    responsive: responsive
  };
}
