// material-ui-responsive-drawer/lib/ResponsiveAppBar.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var _actions = require('./store/actions');

var _selectors = require('./store/selectors');

var _selectors2 = _interopRequireDefault(_selectors);

var _AppBar = require('material-ui/AppBar');

var _AppBar2 = _interopRequireDefault(_AppBar);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ResponsiveAppBar = function (_Component) {
  _inherits(ResponsiveAppBar, _Component);

  function ResponsiveAppBar() {
    _classCallCheck(this, ResponsiveAppBar);

    return _possibleConstructorReturn(this, _Component.apply(this, arguments));
  }

  ResponsiveAppBar.prototype.render = function render() {
    var _props = this.props,
        browser = _props.browser,
        responsiveDrawer = _props.responsiveDrawer,
        breakPoint = _props.breakPoint,
        children = _props.children,
        width = _props.width,
        openSecondary = _props.openSecondary,
        style = _props.style,
        showMenuIconButton = _props.showMenuIconButton,
        onLeftIconButtonClick = _props.onLeftIconButtonClick,
        toggleDrawerOpen = _props.toggleDrawerOpen,
        rest = _objectWithoutProperties(_props, ['browser', 'responsiveDrawer', 'breakPoint', 'children', 'width', 'openSecondary', 'style', 'showMenuIconButton', 'onLeftIconButtonClick', 'toggleDrawerOpen']);

    var props = _extends({}, this.props);
    var setWidth = (0, _selectors2.default)(browser, responsiveDrawer, breakPoint);
    var drawerWidth = width !== undefined ? width : 256;
    var drawerOnRight = openSecondary !== undefined ? openSecondary : false;

    var styles = {
      docked_left: _extends({
        position: 'fixed',
        width: 'auto',
        left: setWidth ? drawerWidth : 0,
        top: 0,
        right: 0
      }, style),
      docked_right: _extends({
        position: 'fixed',
        width: 'auto',
        right: setWidth ? drawerWidth : 0,
        top: 0,
        left: 0
      }, style)
    };

    var appBarProps = _extends({
      width: width,
      style: drawerOnRight ? styles.docked_right : styles.docked_left,
      showMenuIconButton: showMenuIconButton !== undefined ? showMenuIconButton : !setWidth,
      onLeftIconButtonClick: onLeftIconButtonClick !== undefined ? onLeftIconButtonClick : toggleDrawerOpen
    }, rest);

    return _react2.default.createElement(
      _AppBar2.default,
      appBarProps,
      children
    );
  };

  return ResponsiveAppBar;
}(_react.Component);

ResponsiveAppBar.propTypes = process.env.NODE_ENV !== "production" ? {
  responsiveDrawer: _propTypes2.default.object.isRequired,
  browser: _propTypes2.default.object.isRequired,
  toggleDrawerOpen: _propTypes2.default.func.isRequired,
  style: _propTypes2.default.object,
  breakPoint: _propTypes2.default.string,
  width: _propTypes2.default.number,
  openSecondary: _propTypes2.default.bool,
  showMenuIconButton: _propTypes2.default.bool,
  onLeftIconButtonClick: _propTypes2.default.func
} : {};

var mapStateToProps = function mapStateToProps(state) {
  var browser = state.browser,
      responsiveDrawer = state.responsiveDrawer;

  return {
    browser: browser,
    responsiveDrawer: responsiveDrawer
  };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
  return {
    toggleDrawerOpen: function toggleDrawerOpen() {
      dispatch((0, _actions.toggleDrawerOpen)());
    }

  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(ResponsiveAppBar);
module.exports = exports['default'];
